common/nexus/failure.go

```diff
@@ -37,6 +37,15 @@ import (
 	"google.golang.org/protobuf/encoding/protojson"
 )
 
+const (
+	// FailureSourceHeaderName is the header used to indicate from where the Nexus failure originated.
+	FailureSourceHeaderName = "Temporal-Nexus-Failure-Source"
+	// FailureSourceWorker indicates the failure originated from outside the server (e.g. bad request or on the Nexus worker).
+	FailureSourceWorker = "worker"
+)
+
+type FailureSourceContextKey struct{}
```

Comment:
You'll want to create an instance of this struct to use as a unique key.

```suggestion
type FailureSourceContextKeyType struct{}
var FailureSourceContextKey FailureSourceContextKeyType{}
```